### DISCUSSION TITLE
(mongodb) How to run a $search with prisma

### DISCUSSION THREAD

**Discussion by vothvovo** [prisma/prisma#16477] [Q&A] [answered]
I want to run a $search query for atlas search https://www.mongodb.com/docs/atlas/atlas-search/

I think this could be done with $runCommandRaw, but there is almost no documentation regarding it : prisma/docs#3423 

can this even be done with prisma right now?

**Comment by nurul3101** [comment, marked as answer]
Hey @vothvovo 👋 

Yes, you can use `$search` with `$runCommandRaw`, $runCommandRaw accepts all parameters which can be passed in MongoDB's [db.runCommand](https://www.mongodb.com/docs/manual/reference/command/).

Here's an example script which utilizes $search query.

schema.prisma
```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fieldReference"]
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model Author {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  username  String   @unique
  email     String   @unique
  createdAt DateTime @default(now()) @db.Timestamp
}

```

index.ts
```typescript
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({});

async function main() {
  await prisma.author.create({
    data: {
      email: 'bob@prisma.io',
      username: 'Bob',
    },
  });

  const response = await prisma.$runCommandRaw({
    aggregate: 'Author',
    pipeline: [
      {
        $search: {
          near: {
            path: 'createdAt',
            origin: 1669972894,
            pivot: 100000000,
          },
        },
      },
    ],
    cursor: {},
  });

  console.log(JSON.stringify(response, null, 2));
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

Here's the response:
```json
> ts-node index.ts

{
  "cursor": {
    "firstBatch": [],
    "id": 0,
    "ns": "prisma.Author"
  },
  "ok": 1
}

```

**Reply by vothvovo** [reply]
This works, but shouldn't there be some sort of data inside firstBatch ?

**Reply by nurul3101** [reply]
Yes, I just wanted to demonstrate that the query works so that you get a basic idea of how to proceed, the data wasn't exactly accurate in the database, due to which there is no data.

**Reply by vothvovo** [reply]
I know this not part of the original question but,I am getting no data rn. I get an empty response like above, I can assume that my query is somehow wrong, right?

**Reply by nurul3101** [reply]
Yes, I think that would be the case, If it helps I copied the syntax from this part of the documentation: https://www.mongodb.com/docs/atlas/atlas-search/query-syntax/#aggregation-variable

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
